booking-engine-api/operations/reservations.md

```diff
@@ -77,3 +77,165 @@ Get a price quotation for a specific hotel, date interval, room category and per
+    {
+      "Identifier": "uniqueClientGeneratedIdentifier1",
+      "TotalAmount": {
+        "Currency": "EUR",
+        "GrossValue": 203.00,
+        "NetValue": 184.82,
+        "TaxValues": [
+          {
+            "TaxRateCode": "ES-2016-R",
+            "Value": 18.18
+          }
+        ],
+        "Breakdown": {
+          "Items": [
+            {
+              "TaxRateCode": "ES-2016-R",
+              "NetValue": 181.82,
+              "TaxValue": 18.18
+            },
+            {
+              "TaxRateCode": null,
+              "NetValue": 3.0,
+              "TaxValue": 0.0
+            }
+          ]
+        }
+      },
+      "AmountToChargeOnConfirmation": null,
+      "ProductOrderPrices": [
+        {
+          "ProductId": "47c186ec-7b2b-43f2-95fb-abc800c82505",
+          "AgeCategoryId": null,
+          "ProductName": {
+            "es-ES": "Bicicleta",
+            "en-US": "Bike Rental"
+          },
+          "ProductOptions": {
+            "SelectedByDefault": false,
+            "BillAsPackage": false,
+            "OfferToCustomer": true,
+            "ExcludePriceFromOffer": false,
+            "OfferToEmployee": true
+          },
+          "ChargingMode": "Once",
+          "TotalAmount": {
+            "Currency": "EUR",
+            "GrossValue": 3.0,
+            "NetValue": 3.0,
+            "TaxValues": [],
+            "Breakdown": {
+              "Items": [
+                {
+                  "TaxRateCode": null,
+                  "NetValue": 3.0,
+                  "TaxValue": 0.0
+                }
+              ]
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### ReservationPrice
+
+| Property | Type | Contract | Description |
+| :-- | :-- | :-- | :-- |
+| `Identifier` | string | required | Unique identifier for reservation provided in the request data |
+| `AmountToChargeOnConfirmation` | [Amount](#totalamount) | optional | Amount that needs to be charged on payment gateway on reservationGroup confirmation |
+| `TotalAmount` | [Total Amount](#totalamount) | required | `TotalAmount` object of the reservation |
+| `ProductOrderPrices` | array of [ProductOrderPrice](#productorderprice) | required | List of `ProductOrderPrice` objects assigned to reservation |
+
+#### ProductOrderPrice
+| Property | Type | Contract | Description |
+| :-- | :-- | :-- | :-- |
+| `ProductId` | string | required | Unique identifier of product |
+| `AgeCategoryId` | string | optional | Identifier of age category. |
+| `ProductName` | [Localized text](hotels.md#localized-text) | required | Name of the hotel. |
+| `ProductOptions` | [ProductOptions](#productoptions) | required | Product options |
+| `ChargingMode` | string [Product charging mode](hotels.md#product-charging-mode) | required | Charging mode of the product. |
+| `TotalAmount` | [Total Amount](#totalamount) | required | Total amount of product. |
+
+##### ProductOptions
+| Property | Type | Contract | Description |
```

Comment:
I put the Contract back into `Tax Value` but for me, it doesn't make sense to put `Contract` of `optional`/`required` for response values.

I'd say it should be obvious from their type if they're nullable or not.

But let's keep it this way since we have this pattern across whole docs.

Reply:
That's an interesting point @progresak. I assume that `optional` in the response means that the field may or may not be present (so don't be surprised if it's not there), but `required` means you can always expect the field to be present. But I agree, let's follow the standard design pattern for now.

Reply:
I understand, I am thinking more as a developer here because `optional` can be of type `value`, `null`, `undefined`.

_But I just realized that I am not sure if our backend returns `null` and `undefined` or just `null`  - in case of just `null` in all cases, it would be fine. - Will double check_